src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -425,11 +420,15 @@ internal unsafe SocketError DoOperationReceiveMultiBuffer(SafeSocketHandle handl
             NativeOverlapped* overlapped = AllocateNativeOverlapped();
             try
             {
+                int count = _bufferListInternal!.Count;
+                Span<WSABuffer> wsaBuffers = count <= MaxStackWSABuffers ? stackalloc WSABuffer[count] : GC.AllocateUninitializedArray<WSABuffer>(count, pinned: true);
```

Comment:
So if someone passes in more than 16 buffers, we're going to allocate a short-lived temporary on the pinned-object heap?  That doesn't sound great, even if 16 buffers is rare (I would suspect it's rare in the grand scheme of things, but that if an application does it once, it's probably typical for that app.)

Reply:
@Maoni0, per our previous conversations, it'd be better for such an array to be allocated on the normal heap and pinned with fixed (or a GCHandle if necessary, but fixed looks like it'd be sufficient here), correct?

Reply:
Can we array pool here instead.

Reply:
Changed to ArrayPool

Reply:
look like you are now using fixed on a buffer from Arraypool, LGTM.